ray-curator/ray_curator/stages/text/modules/score_filter.py

```diff
@@ -82,10 +115,13 @@ def process(self, batch: DocumentBatch) -> DocumentBatch | None:
 
         """
 
-        score_fn = self.score_fn.score_document if isinstance(self.score_fn, DocumentFilter) else self.score_fn
-
         df = batch.to_pandas()
-        df[self.score_field] = df[self.text_field].apply(score_fn)
+
+        for score_fn_i, text_field_i, score_field_i in zip(
+            self.score_fn, self.text_field, self.score_field, strict=True
+        ):
+            inner_score_fn = score_fn_i.score_document if isinstance(score_fn_i, DocumentFilter) else score_fn_i
+            df[score_field_i] = df[text_field_i].apply(inner_score_fn)
```

Comment:
As a follow up, we may want to just apply a function on text series instead of strings for controlling stuff like batching, etc. 

This is true for both this PR and my PR https://github.com/NVIDIA-NeMo/Curator/pull/898/files which is based on a lot of your work anyways. 

Nothing to do here, just keeping it mind for a follow up PR/Issue of having that option in the base class.

Reply:
Oh interesting, I am curious if this would make a difference in performance? I thought for Pandas it will do it sequentially either way.